### DISCUSSION TITLE
How could I paginate GraphQL calls?

### DISCUSSION THREAD

**Discussion by Brend-Smits** [octokit/octokit.rb#1479] [Q&A] [answered]
Hi all,
I'm looking to retrieve all discussions of a repository using the GraphQL API (there is no REST API available for this information).
I can query the information using Octokit.rb, but I'm having trouble setting up pagination.  

Here's my code:

```Ruby
      query = <<-GRAPHQL
{
  repository(name: "octokit.rb", owner: "octokit") {
    discussions(orderBy: {field: UPDATED_AT, direction: ASC}, first: 1) {
      edges {
        node {
          id
          author {
            login
          }
          title
          number
          comments(first: 1) {
            nodes {
              author {
                login
              }
            }
          }
        }
      }
      pageInfo {
        startCursor
        hasNextPage
        endCursor
      }
      totalCount
    }
  }
}
      GRAPHQL

_client.post('/graphql', { query: query }.to_json)
```
Any feedback is welcome. If this is not easily possible with the library, I will have to fabricate something myself.

**Comment by timrogers** [comment, marked as answer]
👋🏻 Hi @Brend-Smits! We don't have any functionality built into Octokit.rb for GraphQL or especially GraphQL pagination. As you've spotted, you can make a GraphQL request from Octokit, but it doesn't really go any further than that.

I don't have any sample code for that - but in principle, you need to take the `endCursor` returned and pass it in to the `discussions` connection  using the `after` param - and just keep looping until `hasNextPage` is `false`.

I hope that helps! Happy to assist if you run into any issues, and it would be fantastic if you could share any code you wrote.
